agent/fw_lumen.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "php_agent.h"
+#include "php_call.h"
+#include "php_user_instrument.h"
+#include "php_execute.h"
+#include "php_wrapper.h"
+#include "php_hash.h"
+#include "fw_hooks.h"
+#include "util_logging.h"
+#include "util_memory.h"
+#include "util_strings.h"
+
+
+static int nr_lumen_name_the_wt_from_zval(const zval* name TSRMLS_DC,
+                                          const char* lumen_version,
+                                          bool strip_base) {
+  if (nrlikely(nr_php_is_zval_non_empty_string(name))) {
+    char* path = nr_strndup(Z_STRVAL_P(name), Z_STRLEN_P(name));
+    char* action = strrchr(path, '\\') + 1;
+
+    nrl_verbosedebug(NRL_TXN, "path in wt: %s", path);
+
+    nr_txn_set_path(
+        lumen_version, NRPRG(txn), strip_base ? action : path,
+        NR_PATH_TYPE_ACTION,
+        NR_OK_TO_OVERWRITE); /* Watch out: this name is OK to overwrite */
+
+    nr_free(path);
+    return NR_SUCCESS;
+  }
+
+  return NR_FAILURE;
+}
+
+NR_PHP_WRAPPER(nr_lumen_name_the_wt) {
+  zval* route_info = NULL;
+
+  /* Warning avoidance */
+  (void)wraprec;
+
+  /* Verify that we are using Lumen, otherwise bail. */
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_LUMEN);
+
+  /* $routeInfo object used by Application */
+  route_info = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+
+  /* We expect route_info to be an array.  At index 1, if we see an
+   * 'as' key, then we have access to the route, otherwise, if we have
+   * a 'uses' key we have access to the controller and action.
+   * See: https://lumen.laravel.com/docs/5.3/routing#route-group-middleware
+   */
+  if (!nr_php_is_zval_valid_array(route_info)) {
+    nrl_verbosedebug(NRL_TXN, "Lumen: $routeInfo was not an array");
+    NR_PHP_WRAPPER_CALL;
+    goto end;
+  }
+
+  NR_PHP_WRAPPER_CALL;
+
+  /* obtain $routeInfo[1] */
+  zend_ulong idx = 1;
+  zval* route_info_pos
+      = nr_php_zend_hash_index_find(Z_ARRVAL_P(route_info), idx);
```

Comment:
Should we check this for `NULL` before passing it to the next function?

Reply:
All of the `nr_php_zend_hash_` functions are `NULL`-safe or resolve to one that is, but I like this idea of catching it early to avoid extraneous function calls.